code-challenges.test.js

```diff
@@ -15,42 +15,165 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
-
+describe ("secretCodeWordCreator", () => {
+    it ("Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0", () => {
+
+        const secretCodeWord1 = "Lackadaisical"
+        // Expected output: "L4ck4d41s1c4l"
+        const secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k"
+        const secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c"
+        
+        expect (secretCodeWordCreator(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect (secretCodeWordCreator(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect (secretCodeWordCreator(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
 
 // b) Create the function that makes the test pass.
 
+// create a function call secretCodeWordCreator
+// secretCodeWordCreator will have one parameter which will be a string
+// create a variable secretCodeWord
+// secretCodeWord will equal the string.replace
+// string.replace will replace all instances of A,a with 4, E,e with 3, I,i with 0 and O,o with 0
+// return sercetCodeWord
+
+function secretCodeWordCreator (String) {
+    let secretCodeWord = String.replace(/a/g, 4).replace(/A/g, 4).replace(/e/g, 3).replace(/E/g, 3).replace(/i/g, 1).replace(/I/g, 1).replace(/o/g, 0).replace(/O/g, 0)
+    return secretCodeWord
+}
+console.log(secretCodeWordCreator(secretCodeWord1))
+// L4ck4d41s1c4l
+console.log(secretCodeWordCreator(secretCodeWord2))
+// G0bbl3dyg00k
+console.log(secretCodeWordCreator(secretCodeWord3))
+// 3cc3ntr1c
+
```

Comment:
Great work - just for future challenges...when you use regex please explain what it's doing so we know you have an understanding of it.  Well done!